File: RandomData.py
from numpy import random
## CaseNum
CaseLen = int(input("Please input case length: "))
## CaseNum
CaseNum = int(input("Please input case number: "))
## lamba
lamba = input("Please input lambda: ")
##
CarArriveTimes = []
for c in range(CaseNum) :
    TimesCar = random.poisson(lam = float(lamba), size = CaseLen)
    CarArriveTime = [-1]
    for i in range(len(TimesCar)):
         for j in range(TimesCar[i]) :
            CarArriveTime.append(i+1)
    CarArriveTimes.append(CarArriveTime)
###
with open("TestData.txt", "w") as data :
    for c in range(CaseNum) :
        for item in CarArriveTimes[c]:
            data.write(str(item) + " ")
        data.write("\n")

File: DP_approach_to_Optimal_Lane_Merging/RandomData.cpp
#include <iostream>
#include <iomanip>
#include <string>
#include <map>
#include <random>
#include <fstream>
using namespace std;

default_random_engine generator;

int main()
{
	// CaseLen
	cout << "Please input case length: " << endl;
	int CaseLen;
	cin >> CaseLen;
	// CaseNum
	cout << "Please input case number: " << endl;
	int CaseNum;
	cin >> CaseNum;
	// lamba
	cout << "Please input lambda: " << endl;
	float lamba;
	cin >> lamba;
	poisson_distribution<int> distribution(lamba);
	// Do Random and write data
	ofstream TestData;
	TestData.open("TestData.txt");
	for (int i = 0; i < CaseNum; i++)
	{
		vector<int> Arrtime;
		for (int j = 0; j < CaseLen; j++)
		{
			int number = distribution(generator);
			Arrtime.push_back(number);
		}
		vector<int> WriteTime;
		WriteTime.push_back(-1);
		for (int j = 0; j < CaseLen; j++)
		{
			for (int k = 0;k < Arrtime[j];k++)
				WriteTime.push_back(j + 1);
		}
		for (int j = 0; j < WriteTime.size(); j++)
			TestData << WriteTime[j] << " ";
		TestData << endl;
	}
}